dspace-server-webapp/src/main/java/org/dspace/app/rest/authorization/impl/LoginOnBehalfOfFeature.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.authorization.impl;
+
+import java.sql.SQLException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureDocumentation;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.app.rest.model.EPersonRest;
+import org.dspace.app.rest.model.SiteRest;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * The LoginOnBehalfOf feature. It can be used by Administrators when the webui.user.assumelogin property is set to true
+ * to allow the admin to login as another user
+ */
+@Component
+@AuthorizationFeatureDocumentation(name = LoginOnBehalfOfFeature.NAME,
+    description = "It can be used by administrators to login on behalf of a different user")
+public class LoginOnBehalfOfFeature implements AuthorizationFeature {
+
+    public static final String NAME = "loginOnBehalfOf";
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private EPersonService ePersonService;
+
+    @Override
+    public boolean isAuthorized(Context context, BaseObjectRest object) throws SQLException {
+        if (!StringUtils.equals(object.getType(), SiteRest.NAME) &&
+            !StringUtils.equals(object.getType(), EPersonRest.NAME)) {
+            return false;
+        }
+        if (!authorizeService.isAdmin(context)) {
+            return false;
+        }
+        if (!configurationService.getBooleanProperty("webui.user.assumelogin")) {
+            return false;
+        }
+        if (StringUtils.equals(object.getType(), EPersonRest.NAME)) {
+            EPersonRest ePersonRest = (EPersonRest) object;
+            EPerson currentUser = context.getCurrentUser();
+            if (StringUtils.equalsIgnoreCase(currentUser.getEmail(), ePersonRest.getEmail())) {
+                return false;
+            }
+
+            EPerson ePerson = ePersonService.findByEmail(context, ePersonRest.getEmail());
+            if (authorizeService.isAdmin(context, ePerson)) {
+                return false;
```

Comment:
Could we add some comments to this method. It's not well described currently.  It looks like it's covering some scenarios that are not described in the class JavaDocs.  Namely, it looks like you have LoginOnBehalfOf  rights if:
1. Feature is enabled, you request it on a Site and you are an admin
2. Or, Feature is enabled, you request it on an EPerson *and* you are NOT that EPerson and that EPerson is *NOT* and Admin.

Am I reading the code correctly?  The logic here is a bit hard to read, so it'd be good to describe in a comment. or two.